homeassistant/components/homematicip_cloud/event.py

```diff
@@ -86,9 +97,33 @@ async def async_added_to_hass(self) -> None:
     @callback
     def _async_handle_event(self, *args, **kwargs) -> None:
         """Handle the event fired by the functional channel."""
+        raised_channel_event = self._get_channel_event_from_args(*args)
+
+        if not self._should_raise(raised_channel_event):
+            return
+
         event_types = self.entity_description.event_types
         if TYPE_CHECKING:
             assert event_types is not None
 
         self._trigger_event(event_type=event_types[0])
         self.async_write_ha_state()
+
+    def _should_raise(self, event_type: str) -> bool:
+        """Check if the event should be raised."""
+        if self.entity_description.channel_event_types is None:
+            return False
+        return event_type in self.entity_description.channel_event_types
+
+    def _get_channel_event_from_args(self, *args) -> str:
+        """Get the channel event."""
+        if not args:
+            return ""
+
+        if len(args) == 0:
+            return ""
```

Comment:
What's this for?

Reply:
I removed that. I gonna refactor the handler in my upstream lib with next bump. Doesn't make sense to use args.